### PR TITLE
Various CTF fixes

### PR DESCRIPTION
**Pull request by ASpoonPlaysGames**
Fixing the following:
- Various crashes
- Being able to capture flags during epilogue and halftime #266 
- Weird flag spawning stuff

Also I have opted to destroy the flags and the flag bases when the epilogue is triggered, as I believe that's what happens in vanilla? Would be nicer if I could just destroy the actual flags and not the bases, but that means that the UI stays on screen saying to capture the flags

I have no clue why the old flag spawn stuff wasn't working properly, and this new method could use some more testing tbh

### CODE REVIEW

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_ctf.nut [hunk 1]
```diff
@@ -137,14 +140,22 @@ void function CreateFlags()
...
-		bool shouldSwap = SWAP_FLAG_MAPS.contains( GetMapName() ) ? !switchedSides : switchedSides 
-	
+
+		// i dont know why this works and whatever we had before didn't, but yeah
+		bool shouldSwap = switchedSides 
+		if (!shouldSwap && SWAP_FLAG_MAPS.contains( GetMapName() ))
+			shouldSwap = !shouldSwap
+		
+		//printt("Flags will swap: " + shouldSwap)
+
```
Comment (abarichello) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_ctf.nut:150]:
```suggestion
		//printt("Flags will swap: " + shouldSwap)

```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_ctf.nut [hunk 2]
```diff
@@ -137,14 +140,22 @@ void function CreateFlags()
...
 		int flagTeam = spawn.GetTeam()
+		//printt("Flag team before: " + flagTeam)
```
Comment (abarichello) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_ctf.nut:152]:
```suggestion
		//printt("Flag team before: " + flagTeam)
```

File: Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_ctf.nut [hunk 3]
```diff
@@ -137,14 +140,22 @@ void function CreateFlags()
...
 		if ( shouldSwap )
 		{
 			flagTeam = GetOtherTeam( flagTeam )
 			SetTeam( spawn, flagTeam )
 		}
+		//printt("Flag team after: " + flagTeam)
```
Comment (abarichello) [inline Northstar.CustomServers/mod/scripts/vscripts/gamemodes/_gamemode_ctf.nut:158]:
```suggestion
		//printt("Flag team after: " + flagTeam)
```

**Review by abarichello — CHANGES_REQUESTED** [review]

**Review by abarichello — APPROVED** [review]
Ty

**Comment by Revadike** [comment]
Does this mean we finally get some good CTF lobbies? 🤞

**Comment by ASpoonPlaysGames** [comment]
I can't guarantee good, but they should be better

**Pull request merged**